mongo/integration/crud_prose_test.go

```diff
@@ -145,6 +147,25 @@ func TestHintErrors(t *testing.T) {
 	})
 }
 
+type testValueMarshaler struct {
+	val []bson.D
+}
+
+func (tvm testValueMarshaler) MarshalBSONValue() (bsontype.Type, []byte, error) {
+	aidx, arr := bsoncore.AppendArrayStart(nil)
```

Comment:
Instead of creating the array manually, can we do `bson.MarshalValue(tvm.val)`? We're not using custom codecs here, so it should take care of everything automatically.